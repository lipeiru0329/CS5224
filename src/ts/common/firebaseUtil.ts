import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/database';

class FirebaseUtil {
	/**********************************************************
	 *
	 * authentication
	 *
	 **********************************************************/
	private translateFirebaseError(errorCode: string, errorMessage: string): string {
		switch (errorCode) {
			case 'auth/user-not-found':
				return 'Invalid user. Please sign up or log in by other method.';
			case 'auth/wrong-password':
				return 'Invalid password. Please try again.';
			case 'auth/weak-password':
				return 'Password too weak. Please try again.';
			case 'auth/email-already-in-use':
			case 'auth/account-exists-with-different-credential':
				return 'Already signed up. Please log in via the correct method.';
			case 'auth/invalid-email':
				return 'Invalid email. Please try again.';
			default:
				return errorMessage;
		}
	}

	public async emailSignIn(email: string, password: string) {
		try {
			await firebase.auth().signInWithEmailAndPassword(email, password);
			return '';
		} catch (error) {
			return this.translateFirebaseError(error.code, error.message);
		}
	}

	public signOut() {
		return firebase.auth().signOut();
	}
	/**********************************************************
	 *
	 * fetch bot data
	 *
	 **********************************************************/
}

const firebaseUtil = new FirebaseUtil();
export default firebaseUtil;
