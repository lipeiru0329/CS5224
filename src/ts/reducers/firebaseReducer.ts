import { AnyAction } from 'redux';
import * as CST from 'ts/common/constants';
import { IFirebaseState } from 'ts/common/types';

export const initialState = {
	[CST.AC_AUTH]: false
};

export function firebaseReducer(
	state: IFirebaseState = initialState,
	action: AnyAction
): IFirebaseState {
	switch (action.type) {
		case CST.AC_AUTH:
			return action[CST.AC_AUTH]
				? Object.assign({}, state, { [CST.AC_AUTH]: action[CST.AC_AUTH] })
				: initialState;
		default:
			return state;
	}
}
