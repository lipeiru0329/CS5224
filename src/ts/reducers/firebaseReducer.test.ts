import * as CST from 'ts/common/constants';
import { firebaseReducer, initialState } from './firebaseReducer';

describe('firebase reducer', () => {
	let state = initialState;

	test('default', () => {
		state = firebaseReducer(state, { type: 'any' });
		expect(state).toMatchSnapshot();
	});

	test('auth', () => {
		state = firebaseReducer(state, { type: CST.AC_AUTH, [CST.AC_AUTH]: true });
		expect(state).toMatchSnapshot();
	});
});
