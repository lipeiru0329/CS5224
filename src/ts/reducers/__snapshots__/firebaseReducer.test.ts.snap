// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`firebase reducer auth 1`] = `
Object {
  "auth": true,
}
`;

exports[`firebase reducer default 1`] = `
Object {
  "auth": false,
}
`;
